### PR TITLE
WooCommerce - Updated the package modal UI

### LINKED ISSUE #16610: Store shipping: Combine two packages sections into one
Originally came up in: https://github.com/Automattic/wp-calypso/pull/16540. In WCS in core admin, there are two packages sections, breaking up custom and flat rate packages services like USPS provide for free at the post office.

Seeing the two sections in WP.com makes it clear that more work is needed refining this area. The merchant shouldn't have to go through and manage packages two different ways depending on where the package came from and what type it is. 

Here's one solution that takes the existing UI elements and rearranges them a bit to be more cohesive:

Merchant lands on Shipping page and clicks Add Package:

Merchant can add their own custom package:

Or choose from one of the flat rate packages provided by a service

Package has been added:

cc: @marcinbot and @jameskoster for design and implementation feedback.

### PR DESCRIPTION
**Pull request by marcinbot**
Fixes #16610 

Uses input with suffixes for weight and dimension input components
screen shot 2017-08-02 at 16 33 39

Moves the predefined package selection into the modal:
screen shot 2017-08-16 at 14 33 10

To test:
* unit tests should pass
* the package controls should work
* saving should work

### CODE REVIEW

File: client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import FormSectionHeading from 'components/forms/form-section-heading';
+import FormButton from 'components/forms/form-button';
+import inputFilters from './input-filters';
+import PredefinedPackages from './predefined-packages';
+import SegmentedControl from 'components/segmented-control';
+
+const getDialogButtons = ( mode, dismissModal, savePackage, onRemove, translate ) => {
+	const buttons = [
+		<FormButton onClick={ savePackage }>
+			{ ( 'add' === mode ) ? translate( 'Add package' ) : translate( 'Apply changes' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: Hi! I've found a possible matching string that has already been translated **56** times:
`translate( 'Apply Changes' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/16610-packages-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+import FormSectionHeading from 'components/forms/form-section-heading';
+import FormButton from 'components/forms/form-button';
+import inputFilters from './input-filters';
+import PredefinedPackages from './predefined-packages';
+import SegmentedControl from 'components/segmented-control';
+
+const getDialogButtons = ( mode, dismissModal, savePackage, onRemove, translate ) => {
+	const buttons = [
+		<FormButton onClick={ savePackage }>
+			{ ( 'add-custom' === mode ) ? translate( 'Add package' ) : translate( 'Apply changes' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: Hi! I've found a possible matching string that has already been translated **56** times:
`translate( 'Apply Changes' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/16610-packages-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by allendav** [comment]
Taking a look now

File: client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js (hunk 2 above)
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: Hi! I've found a possible matching string that has already been translated **56** times:
`translate( 'Apply Changes' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/16610-packages-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by kellychoffman** [comment]
I accidentally clicked the down arrow and was able to enter a negative dimension:

screen shot 2017-08-16 at 2 33 12 pm

**Comment by kellychoffman** [comment]
Where does the package name appear at checkout? 

screen shot 2017-08-16 at 2 34 59 pm

**Comment by kellychoffman** [comment]
Can the action button be the same for both service package and custom package? Right now, service package says "Apply changes". They should both say 'Add package' or 'Add packages' depending on how many are chosen.

File: client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js (hunk 2 above)
Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: Hi! I've found a possible matching string that has already been translated **56** times:
`translate( 'Apply Changes' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/16610-packages-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (a8ci18n) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: Hi! I've found a possible matching string that has already been translated **56** times:
`translate( 'Apply Changes' )` *ES Score: 15*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/16610-packages-ui)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

**Comment by jameskoster** [comment]
> What should it say if someone uses the checkboxes to remove some of the service packages or both add and remove service packages?

"Done", like payments :)

**Comment by kellychoffman** [comment]
> So is it three possible labels: "Add package" for custom package or one added service package, plural for multiple service packages and "Done" if it's a combination of removal and addition?

Forgot that you could remove a package. Let's go with your suggestion above.

File: client/extensions/woocommerce/woocommerce-services/state/action-types.js [hunk 3]
```diff
@@ -1,16 +1,19 @@
...
+export const WOOCOMMERCE_SERVICES_PACKAGES_ADD_PACKAGE = 'WOOCOMMERCE_SERVICES_PACKAGES_ADD_PACKAGE';
+export const WOOCOMMERCE_SERVICES_PACKAGES_REMOVE_PACKAGE = 'WOOCOMMERCE_SERVICES_PACKAGES_REMOVE_PACKAGE';
+export const WOOCOMMERCE_SERVICES_PACKAGES_EDIT_PACKAGE = 'WOOCOMMERCE_SERVICES_PACKAGES_EDIT_PACKAGE';
+export const WOOCOMMERCE_SERVICES_PACKAGES_DISMISS_MODAL = 'WOOCOMMERCE_SERVICES_PACKAGES_DISMISS_MODAL';
+export const WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_SAVING = 'WOOCOMMERCE_SERVICES_PACKAGES_SET_IS_SAVING';
+export const WOOCOMMERCE_SERVICES_PACKAGES_SET_MODAL_ERRORS = 'WOOCOMMERCE_SERVICES_PACKAGES_SET_MODAL_ERROR';
+export const WOOCOMMERCE_SERVICES_PACKAGES_SAVE_PACKAGE = 'WOOCOMMERCE_SERVICES_PACKAGES_SAVE_PACKAGE';
+export const WOOCOMMERCE_SERVICES_PACKAGES_UPDATE_PACKAGES_FIELD = 'WOOCOMMERCE_SERVICES_PACKAGES_UPDATE_PACKAGES_FIELD';
+export const WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_OUTER_DIMENSIONS = 'WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_OUTER_DIMENSIONS';
+export const WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_ALL = 'WOOCOMMERCE_SERVICES_PACKAGES_TOGGLE_ALL';
```
Comment (allendav) [inline client/extensions/woocommerce/woocommerce-services/state/action-types.js]: The name of this action is kinda general. Toggle all what?

File: client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	};
+
+	const onClose = () => ( dismissModal( siteId ) );
+	const onRemove = () => removePackage( siteId, index );
+
+	const switchMode = ( option ) => {
+		setAddMode( siteId, option.value );
+	};
+
+	const heading = 'edit' === mode ? translate( 'Edit package' ) : translate( 'Add a package' );
```
Comment (allendav) [inline client/extensions/woocommerce/woocommerce-services/views/packages/package-dialog.js]: You use add-predefined === mode and edit === mode and add-custom === mode quite a few times here. Please consider defining intermediate variables to increase readability and avoid breakage down the road if someone misses one.

**Pull request merged**